src/bootstrap/check.rs

```diff
@@ -106,7 +106,8 @@ pub fn cargo(build: &Build, stage: u32, host: &str) {
     let ref newpath = format!("{}{}{}", path.display(), sep, old_path);
 
     let mut cargo = build.cargo(compiler, Mode::Tool, host, "test");
-    cargo.arg("--manifest-path").arg(build.src.join("src/tools/cargo/Cargo.toml"));
+    cargo.arg("--manifest-path").arg(build.src.join("src/Cargo.toml"));
+    cargo.arg("--project").arg("cargo");
```

Comment:
I think in theory this shouldn't need to be modified, but did you find it did need to be?